### PR TITLE
iox-#1755 Distinct overloads for arithmetic types for logstream

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR adds distinct overloads for `LogStream& operator <<`. This enabled implicit type conversion to the underlying type of e.g. `Wrapper<T>` if the implicit `operator T()` is implemented.

Additionally, it is finally possible to log `char` instead of only `signed char` and `unsigned char`.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1755

### CODE REVIEW

File: iceoryx_hoofs/reporting/include/iox/log/logstream.hpp [hunk 1]
```diff
@@ -156,17 +157,84 @@ class LogStream
...
+
+    /// @brief Logging support for 'short'
+    /// @param[in] val is the 'short' to log
     /// @return a reference to the LogStream instance
-    template <typename T, typename std::enable_if_t<std::is_arithmetic<T>::value, bool> = 0>
-    LogStream& operator<<(const T val) noexcept;
+    LogStream& operator<<(const short val) noexcept;
+
+    /// @brief Logging support for 'unsigned shor'
+    /// @param[in] val is the 'unsigned shor' to log
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/reporting/include/iox/log/logstream.hpp:188]:
```suggestion
    /// @brief Logging support for 'unsigned short'
    /// @param[in] val is the 'unsigned short' to log
```

**Review by FerdinandSpitzschnueffler — APPROVED** [review]
I found some typos, but you can ignore my comment and merge if there's nothing else to change.

File: iceoryx_hoofs/reporting/include/iox/log/logstream.hpp (hunk 1 above)
Comment (elBoberido) [reply iceoryx_hoofs/reporting/include/iox/log/logstream.hpp]: fixed

File: iceoryx_hoofs/test/moduletests/test_reporting_logstream.cpp [hunk 2]
```diff
@@ -128,6 +129,45 @@ TEST_F(IoxLogStream_test, StreamOperatorStdString)
     EXPECT_THAT(loggerMock.logs[1].message, StrEq(constLogValue));
 }
 
+TEST_F(IoxLogStream_test, StreamOperatorChar)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2a1fff17-e388-4f84-bb16-30bb3432ae9d");
+    char logValue{'b'};
+    const char constLogValue{'o'};
+    constexpr char constexprLogValue{'b'};
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/moduletests/test_reporting_logstream.cpp:137]: Nitpick

```suggestion
    constexpr char CONSTEXPR_LOG_VALUE{'b'};
```

File: iceoryx_hoofs/test/moduletests/test_reporting_logstream.cpp [hunk 3]
```diff
@@ -436,6 +498,36 @@ TYPED_TEST(IoxLogStreamArithmetic_test, StreamOperator_ValueMax)
     EXPECT_THAT(this->loggerMock.logs[0].message, StrEq(convertToString(this->LogValueMax)));
 }
 
+TYPED_TEST(IoxLogStreamArithmetic_test, StreamOperator_WrappedAndImplicitelyConverted_ValueLow)
```
Comment (mossmaurice) [inline iceoryx_hoofs/test/moduletests/test_reporting_logstream.cpp:501]: Nitpick, also in every other test name.

```suggestion
TYPED_TEST(IoxLogStreamArithmetic_test, StreamOperator_WrappedAndimplicitlyConverted_ValueLow)
```

**Review by mossmaurice — APPROVED** [review]
LGTM, only minor nitpicks. From my side we can merge this.

**Review by mossmaurice — APPROVED** [review]

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Pull request merged**